05week/spaceTravelToMars.js

```diff
@@ -9,7 +9,109 @@ let jobTypes = {
   programmer: 'Any Ship!'
 };
 
+console.log(jobTypes)
 // Your code here
+//build two classes, 
+//crew member
+  //constructor should take in as input: name, their job, their special skill 
+
+class CrewMember {
+  constructor(name, job, specialSkill) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+
+    //we need to assign this information back the crew member 
+    this.ship = null;
+      //unlike ship, we're only going to have one ship assign to one individual
+      // console.log(`crew member's name: ${name}, their job: ${job}, their skill: ${specialSkill}, assigned to ship: ${this.postedToShip}`) 
+    } 
+//create a method should add THIS crew member to the ship being passed i
+      //shipName is the instance of the template you are using. Since you are assigning this information by the variable name, it should pass the whole instance into this. 
+      enterShip(ship){
+        // console.log(`inside my enter ship method? ${ship}`)
+        // pass the ship name to the individual 
+           //NOTE: an entire ship instance is passed in, not just a name
+            //need to figure out a way to remove the crew key
+          this.ship = ship
+          //NOTE: the entire crew number should be added to the ship's array of crew 
+            //since we are passing in the ship name, it should be able to access the crew array to had this individual  
+          ship.crew.push(this)
+            //the this being the name on given to the specific crew member. 
+
+        }
+        
+  }
+
+//ship 
+  //constructor should take in as input:name, type, ability
+    //can have crew members 
+
+class Ship {
+  constructor(shipName, type, ability) {
+
+    this.shipName = shipName;
+    this.type = type;
+    this.ability = ability
+
+    // console.log(`name of Ship: ${shipName}, type ${type}, ability ${ability}`)
+
+    //need to have an empty array for crew members
+    this.crew = [
+        //we need to push the whole class on the member in here. 
+        //once push, we can access the whole object of that crew member here. 
+        //see enterShip() in CrewMember to see how they come to be in this list 
+    ]
+
+    //we need to stop the crew from including the ship information
+    // console.log(`currently on the ship? ${this.crew}`)
+  }
+
+      //contain a method: missionStatement()
+  missionStatement() {
+      //is there any one on the ship?
+      if(this.crew.length == 0) {
+        //if not, then we're not going anywhere
+        return "Can't perform a mission yet."
+      } 
+  
+      else {
+        //was told by tutor I did not need to include this, but then in other conversations thought I did. Doesn't pass the test, but doesn't seem like the test was designed for this. 
```

Comment:
You do need some logic to check that the ship's type matches up with the crew member's job.

The commented out code was a good start, but not all the way there yet